internal/bucket/lifecycle/expiration.go

```diff
@@ -186,6 +187,11 @@ func (e Expiration) Validate() error {
 		return errLifecycleInvalidExpiration
 	}
 
+	// DeleteAll set without expiration days
+	if e.DeleteAll.set && e.IsDaysNull() {
```

Comment:
Does DeleteAll with a date instead of expiry will not work ?

Reply:
Correct @vadmeste